### ISSUE TITLE
`addTypename: true` causes webpack to crash

### ISSUE THREAD

**Issue by benjie** [sysgears/persistgraphql-webpack-plugin#3]
```
/Users/benjiegillam/Documents/Mindtree/mtree/node_modules/lodash/lodash.js:214
      rsNonCharRange = '\\x00-\\x2f\\x3a-\\x40\\x5b-\\x60\\x7b-\\xbf',
                               ^

RangeError: Maximum call stack size exceeded
    at isBuffer (buffer.js:281:12)
    at baseClone (/Users/benjiegillam/Documents/Mindtree/mtree/node_modules/lodash/lodash.js:2666:13)
    at /Users/benjiegillam/Documents/Mindtree/mtree/node_modules/lodash/lodash.js:2702:34
    at arrayEach (/Users/benjiegillam/Documents/Mindtree/mtree/node_modules/lodash/lodash.js:537:11)
    at baseClone (/Users/benjiegillam/Documents/Mindtree/mtree/node_modules/lodash/lodash.js:2696:7)
    at /Users/benjiegillam/Documents/Mindtree/mtree/node_modules/lodash/lodash.js:2702:34
    at arrayEach (/Users/benjiegillam/Documents/Mindtree/mtree/node_modules/lodash/lodash.js:537:11)
    at baseClone (/Users/benjiegillam/Documents/Mindtree/mtree/node_modules/lodash/lodash.js:2696:7)
    at /Users/benjiegillam/Documents/Mindtree/mtree/node_modules/lodash/lodash.js:2702:34
    at arrayEach (/Users/benjiegillam/Documents/Mindtree/mtree/node_modules/lodash/lodash.js:537:11)
    at baseClone (/Users/benjiegillam/Documents/Mindtree/mtree/node_modules/lodash/lodash.js:2696:7)
    at /Users/benjiegillam/Documents/Mindtree/mtree/node_modules/lodash/lodash.js:2702:34
    at arrayEach (/Users/benjiegillam/Documents/Mindtree/mtree/node_modules/lodash/lodash.js:537:11)
    at baseClone (/Users/benjiegillam/Documents/Mindtree/mtree/node_modules/lodash/lodash.js:2696:7)
    at /Users/benjiegillam/Documents/Mindtree/mtree/node_modules/lodash/lodash.js:2702:34
    at arrayEach (/Users/benjiegillam/Documents/Mindtree/mtree/node_modules/lodash/lodash.js:537:11)
```

I couldn't figure it out.

**Comment by larixer** [comment]
@benjie Yeah, I was looking at this today. There is some cosmic crap happens in this line:
https://github.com/apollographql/persistgraphql/blob/38814ba5770de28f7dbd5d36d7aa1398be264a76/src/queryTransformers.ts#L63
When I replace this line with:
``` js
  const docClone = cloneDeep(JSON.parse(JSON.stringify(doc)));
```
then this maximum stack exceeded error doesn't happen.

But I'm still don't know why this happens and how to fix it. I am going to look into this tomorrow too.

**Comment by larixer** [comment]
@benjie I have added work around this problem into this plugin. Before calling addTypename we do JSON.parse(JSON.stringify(parsedDocument)), this somehow sanitizes parsed graphql document and lodash cloneDeep doesn't consume too much stack. The workaround is included into version `0.2.1` of the plugin

Of course to fully understand the root of the problem someone needs to examine the structure of returned object value by `graphql` -> `parse` function and find out why `cloneDeep` consumes too much stack when trying to clone it.

**Comment by larixer** [comment]
@benjie I have just published `0.2.2`. `0.2.1` introduced a problem with webpack crashing on source map generation, which have been resolved in `0.2.2`.
